pkg/sentry/socket/externalstack/socket.go

```diff
@@ -0,0 +1,199 @@
+	return 0, nil, 0, nil
+}
+
+func (s *socketOperations) Connect(t *kernel.Task, sockaddr []byte, blocking bool) *syserr.Error {
+	//TODO: implement glue layer
+	return nil
+}
+
+func (s *socketOperations) Shutdown(t *kernel.Task, how int) *syserr.Error {
+	//TODO: implement glue layer
+	return nil
+}
+
+func (s *socketOperations) GetSockOpt(t *kernel.Task, level int, name int, outPtr hostarch.Addr, outLen int) (marshal.Marshallable, *syserr.Error) {
+	//TODO: implement glue layer
+	return nil, nil
+}
+
+func (s *socketOperations) SetSockOpt(t *kernel.Task, level int, name int, optVal []byte) *syserr.Error {
+	//TODO: implement glue layer
+	return nil
+}
+
+func (s *socketOperations) RecvMsg(t *kernel.Task, dst usermem.IOSequence, flags int, haveDeadline bool, deadline ktime.Time, senderRequested bool, controlDataLen uint64) (int, int, linux.SockAddr, uint32, socket.ControlMessages, *syserr.Error) {
+	//TODO: implement glue layer
+	return 0, 0, nil, 0, socket.ControlMessages{}, syserr.ErrInvalidArgument
+}
+
+func (s *socketOperations) SendMsg(t *kernel.Task, src usermem.IOSequence, to []byte, flags int, haveDeadline bool, deadline ktime.Time, controlMessages socket.ControlMessages) (int, *syserr.Error) {
+	//TODO: implement glue layer
+	return 0, nil
+}
+
+func (s *socketOperations) State() uint32 {
+	//TODO: implement glue layer
+	return 0
+}
+
+func (s *socketOperations) Type() (family int, skType linux.SockType, protocol int) {
+	//TODO: implement glue layer
+	return 0, 0, 0
+}
+
+func (s *socketOperations) OnClose(ctx context.Context) error {
+	return nil
+}
+
+func (s *socketOperations) EventRegister(e *waiter.Entry) error {
+	//TODO: implement glue layer
+	return nil
+}
+
+func (s *socketOperations) EventUnregister(e *waiter.Entry) {
+	//TODO: implement glue layer
+}
+
+func (s *socketOperations) Readiness(mask waiter.EventMask) waiter.EventMask {
+	//TODO: implement glue layer
+	return 0
+}
+
+func (s *socketOperations) Read(ctx context.Context, dst usermem.IOSequence, opts vfs.ReadOptions) (int64, error) {
+	//TODO: implement glue layer
+	return 0, nil
+}
+
+func (s *socketOperations) Write(ctx context.Context, src usermem.IOSequence, opts vfs.WriteOptions) (int64, error) {
+	//TODO: implement glue layer
+	return 0, nil
+}
+
+func (s *socketOperations) Epollable() bool {
+	return true
+}
+
+func (s *socketOperations) Ioctl(ctx context.Context, io usermem.IO, sysno uintptr, args arch.SyscallArguments) (uintptr, error) {
+	//TODO: implement glue layer
+	return 0, nil
+}
+
+func (s *socketOperations) Release(ctx context.Context) {
+	//TODO: implement glue layer
+}
+
+func (s *socketOperations) GetSockName(t *kernel.Task) (linux.SockAddr, uint32, *syserr.Error) {
+	//TODO: implement glue layer
+	return nil, 0, nil
+}
+
+func (s *socketOperations) GetPeerName(t *kernel.Task) (linux.SockAddr, uint32, *syserr.Error) {
+	//TODO: implement glue layer
+	return nil, 0, nil
+}
+
+func (s *socketOperations) saveTimestamp() int64 {
```

Comment:
This disappeared. Was it supposed to?

Reply:
Indeed, we removed saveTimestamp and loadTimestamp for plugin netstack. We thought it may not be necessary for plugin netstack to support checkpoint-restore feature. We just dropped this two functions for now. However, if you have a different perspective, please do let us know. We are open to considering alternative viewpoints.